filters/include/pcl/filters/experimental/voxel_grid.h

```diff
@@ -101,52 +102,47 @@ class VoxelStruct {
   using PointCloudPtr = typename PointCloud::Ptr;
   using PointCloudConstPtr = typename PointCloud::ConstPtr;
   using Grid = typename std::unordered_map<std::size_t, VoxelT>;
-  using GridIterator = typename Grid::iterator;
+  using iterator = typename Grid::iterator;
 
   /** \brief Constructor.
-   * \param[in] transform_filter pointer to the TransformFilter object
    */
-  VoxelStruct(TransformFilter<VoxelStruct> const* transform_filter)
-  {
-    filter_name_ = "VoxelGrid";
-    grid_filter_ = static_cast<CartesianFilter<VoxelStruct> const*>(transform_filter);
-  }
-
-  /** \brief Empty constructor. */
-  VoxelStruct() {}
+  VoxelStruct() { filter_name_ = "VoxelGrid"; }
 
   /** \brief Empty destructor. */
   ~VoxelStruct() {}
 
   /** \brief Get the number of voxels in the grid. */
-  std::size_t
+  inline std::size_t
   size() const
   {
     return grid_.size();
   }
 
   /** \brief Get the begin iterator of the grid. */
-  GridIterator
+  inline iterator
   begin()
   {
     return grid_.begin();
   }
 
   /** \brief Get the end iterator of the grid. */
-  GridIterator
+  inline iterator
   end()
   {
     return grid_.end();
   }
 
   /** \brief Set up the voxel grid variables needed for filtering.
+   *  \param[in] transform_filter pointer to the TransformFilter object
    */
-  bool
-  setUp()
+  inline bool
+  setUp(TransformFilter<Filter, VoxelStruct> const* transform_filter)
```

Comment:
```suggestion
  template <class FilterT=pcl::Filter>
  inline bool
  setUp(TransformFilter<FilterT, VoxelStruct> const* transform_filter)
```

Thoughts on making the setup templated for the voxel struct?

Reply:
Done, I think it's nice to allow reusing `VoxelStruct`